tests/table/binary/test_fixed_size_binary_concat.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import annotations
+
+import pyarrow as pa
+import pytest
+
+from daft.expressions import col, lit
+from daft.table import MicroPartition
+
+
+@pytest.mark.parametrize(
+    "input_data1,input_data2,expected_result,size1,size2",
+    [
+        # Basic concatenation
+        ([b"abc", b"def", b"ghi"], [b"xyz", b"uvw", b"rst"], [b"abcxyz", b"defuvw", b"ghirst"], 3, 3),
+        # With nulls
+        ([b"ab", None, b"cd"], [b"xy", b"uv", None], [b"abxy", None, None], 2, 2),
+        # Special sequences
+        (
+            [b"\x00\x01", b"\xff\xfe", b"\xe2\x98"],
+            [b"\x99\x00", b"\x01\xff", b"\xfe\xe2"],
+            [b"\x00\x01\x99\x00", b"\xff\xfe\x01\xff", b"\xe2\x98\xfe\xe2"],
+            2,
+            2,
+        ),
+        # Complex UTF-8 sequences
+        (
+            [b"\xe2\x98", b"\xf0\x9f", b"\xf0\x9f"],  # Partial UTF-8 sequences
+            [b"\x83\x00", b"\x98\x89", b"\x8c\x88"],  # Complete the sequences
+            [b"\xe2\x98\x83\x00", b"\xf0\x9f\x98\x89", b"\xf0\x9f\x8c\x88"],  # Complete UTF-8 characters
+            2,
+            2,
+        ),
+        # Mixed length concatenation
+        (
+            [b"a", b"b", b"c", b"d"],  # Single bytes
+            [b"12", b"34", b"56", b"78"],  # Two bytes
+            [b"a12", b"b34", b"c56", b"d78"],  # Three bytes result
+            1,
+            2,
+        ),
+        # Zero bytes in different positions
+        (
+            [b"\x00a", b"a\x00", b"\x00\x00"],  # Zeros in different positions
+            [b"b\x00", b"\x00b", b"c\x00"],  # More zeros
+            [b"\x00ab\x00", b"a\x00\x00b", b"\x00\x00c\x00"],  # Combined zeros
+            2,
+            2,
+        ),
+    ],
+)
+def test_fixed_size_binary_concat(
+    input_data1: list[bytes | None],
+    input_data2: list[bytes | None],
+    expected_result: list[bytes | None],
+    size1: int,
+    size2: int,
+) -> None:
+    table = MicroPartition.from_pydict(
+        {
+            "a": pa.array(input_data1, type=pa.binary(size1)),
+            "b": pa.array(input_data2, type=pa.binary(size2)),
+        }
+    )
+    # Verify inputs are FixedSizeBinary before concatenating
+    assert str(table.schema()["a"].dtype) == f"FixedSizeBinary[{size1}]"
+    assert str(table.schema()["b"].dtype) == f"FixedSizeBinary[{size2}]"
+    result = table.eval_expression_list([col("a").binary.concat(col("b"))])
+    assert result.to_pydict() == {"a": expected_result}
+    # Result should be FixedSizeBinary with combined size when both inputs are FixedSizeBinary
+    assert str(result.schema()["a"].dtype) == f"FixedSizeBinary[{size1 + size2}]"
```

Comment:
you can construct daft.DataType values and compare them directly instead of comparing strings